### ISSUE TITLE
FlatCompat not working anymore with eslint v9

### ISSUE THREAD

**Issue by codeflorist** [eslint/eslintrc#157]
I was using this solution to import the tailwindcss-plugin into the flat-config:
https://github.com/antfu/eslint-config/issues/315#issuecomment-1799052876

this was working with eslint v8, but with v9 it is not working anymore. The tailwindcss-rules don't seem to get applied.

**Comment by codeflorist** [comment]
@nzakas

Sure, no problem!

Here is a stackblitz project with eslint v8.57.0:
https://stackblitz.com/edit/nuxt-starter-jhkmqy?file=app.vue

It correctly displays the following warnings from `eslint-plugin-tailwindcss`:

```
/home/projects/nuxt-starter-jhkmqy/app.vue
  2:8  warning  Invalid Tailwind CSS classnames order  tailwindcss/classnames-order
```

Here is the same project with eslint v9.1.1:
https://stackblitz.com/edit/nuxt-starter-ncst5t?file=app.vue

It does not display the warnings.

(I don't know why the v9 stackblitz project is throwing all those peer dependency warnings. I doesn't happen in my real-world projects.)

The stackblitz examples use `antfu/eslint-config` with Nuxt 3. Using `nuxt/eslint` has the same problem.

Many thanks for taking a look at it and your general work on eslint!

**Comment by nzakas** [comment]
I don't think this has anything to do with the `@eslint/eslintrc` package. That is the same in both of your test cases. It's more likely this is a problem with ESLint itself, so moving to that repo.

Interestingly, it appears the configuration is correctly calculated in both cases. If you run this in both cases, you get the same result:

```shell
npx eslint --print-config app.vue
```

And if I run `npx eslint app.vue --debug`, it appears that the file is actually linted successfully.

**My hunch:** I think ESLint v9.0.0 might actually be generating a warning message but just not displaying it on the CLI.

Unfortunately, I'm offline for the next few days, but maybe someone else on @eslint/eslint-team can take a look.
